### PR TITLE
Enable cpp code path for `Categorify` ops

### PR DESCRIPTION
**Pull request by rjzamora**
Rolls back one of the changes in https://github.com/NVIDIA-Merlin/systems/pull/354 to improve inference performance of `Categorify`.

### CODE REVIEW

File: merlin/systems/workflow/base.py [hunk 1]
```diff
@@ -66,15 +66,18 @@ def __init__(self, workflow, output_dtypes, model_config, model_device):
...
-        # (disabled for now while we sort out whether and how we want to use C++ implementations
-        # of NVTabular operators for performance optimization)
-        # self._initialize_ops(self.workflow.output_node)
+        # (disabled everyting other than Categorify for now while we sort out whether
+        # and how we want to use C++ implementations of NVTabular operators for
+        # performance optimization)
+        self._initialize_ops(self.workflow.output_node, restrict=["Categorify"])
 
-    def _initialize_ops(self, workflow_node, visited=None):
+    def _initialize_ops(self, workflow_node, visited=None, restrict=False):
```
Comment (oliverholworthy) [inline merlin/systems/workflow/base.py:74]: perhaps could change this to `restrict=None` and set `restrict = restrict or []` in the method to keep the type consistent as an optional list of node names

Comment (rjzamora) [inline merlin/systems/workflow/base.py:74]:
```suggestion
    def _initialize_ops(self, workflow_node, visited=None, restrict=None):
        restrict = restrict or []

```

**Review by oliverholworthy — APPROVED** [review]

**Comment by oliverholworthy** [comment]
the docs related errors are now resolved. 

There's two remaining things for this MR:
- fix the lint errors (reformat with black `pre-commit run --all-files black`) and the codespell error (typo in everything)
- merge the NVTabular PR https://github.com/NVIDIA-Merlin/NVTabular/pull/1874
  - currently blocked by some unrelated errors with docs (which will be resolved by https://github.com/NVIDIA-Merlin/NVTabular/pull/1878 )

**Comment by rjzamora** [comment]
>There's two remaining things for this MR: ...

Thanks so much @oliverholworthy ! Sorry, I missed this earlier.

**Comment by oliverholworthy** [comment]
Following the merge of https://github.com/NVIDIA-Merlin/NVTabular/pull/1874 we're now getting a new error in  the tests

```
RuntimeError: Error: <class 'ValueError'> - Unknown column for CategorifyTransform x__values
```

**Comment by rjzamora** [comment]
I haven't had the time to understand the test failure well enough to work through a long-term fix. However, my last commit roles back the decision to use the cpp code path for "Categorify" by default. Instead, when the `WorkflowRunner` is initialized, the value of the `NVT_CPP_OPS` environment variable will dictate which NVT ops the cpp code path will be used for.

For example, you can enable the cpp version of `Categorify` by deploying a triton model as follows:

```
NVT_CPP_OPS="Categorify" tritonserver --model-repository=/my-model/
```

**Comment by jperez999** [comment]
This is failing right now on some package mismatches right now. A solution is to downgrade the tritonclient package to 2.28.0. Working on that solution now. Then we should be able to pick up the newest version of CI container to fix the triton errors.

File: merlin/systems/workflow/base.py [hunk 2]
```diff
@@ -66,15 +67,24 @@ def __init__(self, workflow, output_dtypes, model_config, model_device):
...
 
         # recurse over all column groups, initializing operators for inference pipeline.
-        # (disabled for now while we sort out whether and how we want to use C++ implementations
-        # of NVTabular operators for performance optimization)
-        # self._initialize_ops(self.workflow.output_node)
+        # (disabled everything other than operators that are specifically listed
+        # by the `NVT_CPP_OPS` environment variable while we sort out whether
+        # and how we want to use C++ implementations of NVTabular operators for
+        # performance optimization)
+        _nvt_cpp_ops = os.environ.get("NVT_CPP_OPS", "").split(",")
```
Comment (rjzamora) [inline merlin/systems/workflow/base.py:74]:
```suggestion
        _nvt_cpp_ops = os.environ.get("NVT_CPP_OPS", "Categorify").split(",")
```

**Pull request merged**